pennylane/operation.py

```diff
@@ -1385,6 +1385,55 @@ def compute_qfunc_decomposition(*args, **hyperparameters) -> None:
 
         raise DecompositionUndefinedError
 
+    @classproperty
+    def resource_keys(self) -> tuple:
+        """The set of parameters that affects the resource requirement of the operator.
+
+        All decomposition rules for this operator class are expected to have a resource function
+        that accepts keyword arguments that match these keys exactly. The :func:`~pennylane.resource_rep`
+        function will also expect keyword arguments that match these keys when called with this
+        operator type.
+
+        .. seealso::
+            :meth:`~.Operator.resource_params`
+
+        """
```

Comment:
Should we either raise an `NotImplementedError` or return an empty tuple?

Reply:
I don't think you can raise an error from a `@classproperty` without CI failing. I'll have it return an empty set (yes it's a set, thanks for catching this)